stripe/src/test/java/com/stripe/android/StripePaymentAuthTest.kt

```diff
@@ -116,6 +121,50 @@ class StripePaymentAuthTest {
         verify(paymentController).handleSetupResult(data, setupCallback)
     }
 
+    @Test
+    fun confirmAlipayPayment_shouldConfirmAndAuth() {
+        val authenticationCallbackCaptor: KArgumentCaptor<ApiResultCallback<StripeIntent>> = argumentCaptor()
+
+        val stripe = createStripe()
+        val confirmPaymentIntentParams = ConfirmPaymentIntentParams.createAlipay(
+            "client_secret",
+            "yourapp://post-authentication-return-url")
+        val authenticationHandler = object : AlipayAuthenticator {
+            override fun onAuthenticationRequest(data: String): Map<String, String> {
+                return mapOf("resultStatus" to "9000")
+            }
+        }
+        stripe.confirmAlipayPayment(
+            confirmPaymentIntentParams,
+            authenticator = authenticationHandler,
+            callback = paymentCallback
+        )
+        verify(paymentController).startConfirm(
+            eq(confirmPaymentIntentParams),
+            eq(REQUEST_OPTIONS),
+            authenticationCallbackCaptor.capture()
+        )
+        val authenticationCallback = authenticationCallbackCaptor.firstValue
+
+        // passes through errors
+        verify(paymentCallback, never()).onError(any())
+        val e = RuntimeException()
+        authenticationCallback.onError(e)
+        verify(paymentCallback).onError(e)
+
+        // handles authentication
+        val confirmedIntent = PaymentIntentJsonParser().parse(
+            PaymentIntentFixtures.ALIPAY_REQUIRES_ACTION_JSON
+        )!!
```

Comment:
you may want to create a parsed `PaymentIntent` instance in `PaymentIntentFixtures` (e.g. `PaymentIntentFixtures.ALIPAY_REQUIRES_ACTION`)